gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/ciba/resources/handler/AuthenticationRequestParametersHandler.java

```diff
@@ -0,0 +1,122 @@
+        this.domain = domain;
+        this.jwsService = jwsService;
+        this.jwkService = jwkService;
+        this.userService = userService;
+
+        this.bindingMessageMaxLength = domain.getOidc().getCibaSettings().getBindingMessageLength();
+    }
+
+    @Override
+    public void handle(RoutingContext context) {
+        final CibaAuthenticationRequest request = createCibaRequest(context);
+
+        final OpenIDProviderMetadata openIDProviderMetadata = context.get(PROVIDER_METADATA_CONTEXT_KEY);
+        final Client client = context.get(CLIENT_CONTEXT_KEY);
+
+        validateScopes(request);
+
+        validateAcrValue(openIDProviderMetadata, request);
+
+        validateHints(request);
+
+        validateBindingMessage(request);
+
+        validateUserCode(client, request);
+
+        new CibaAuthenticationRequestResolver(domain, jwsService, jwkService, userService)
+                .resolve(request, client)
+                .subscribe(requestValidated -> {
+                    context.put(CIBA_AUTH_REQUEST_KEY, requestValidated);
+                    context.next();
+                }, context::fail);
+    }
+
+    protected CibaAuthenticationRequest createCibaRequest(RoutingContext context) {
+        return CibaAuthenticationRequest.createFrom(context);
+    }
+
+    private void validateScopes(CibaAuthenticationRequest request) {
+        if (request.getScopes() == null || !request.getScopes().contains(Scope.OPENID.getKey())) {
+            throw new InvalidScopeException("scope is missing or doesn't contain 'openid'");
+        }
+    }
+
+    private void validateAcrValue(OpenIDProviderMetadata openIDProviderMetadata, CibaAuthenticationRequest request) {
+        if (request.getAcrValues() != null && !request.getAcrValues().stream().anyMatch(openIDProviderMetadata.getAcrValuesSupported()::contains)) {
+            throw new InvalidRequestException("Unsupported acr values");
+        }
+    }
+
+    private void validateHints(CibaAuthenticationRequest request) {
+        final long hints = Stream.of(request.getLoginHint(), request.getLoginHintToken(), request.getIdTokenHint())
+                .filter(value -> !StringUtils.isEmpty(value))
+                .count();
+        if (hints != 1) {
+            throw new InvalidRequestException("Only one of login_hint_token, id_token_hint, login_hint is accepted");
+        }
+    }
+
+    private void validateUserCode(Client client, CibaAuthenticationRequest request) {
+        if (StringUtils.isEmpty(request.getUserCode()) && client.getBackchannelUserCodeParameter()) {
+            throw new MissingUserCodeException("user_code is missing");
+        }
+    }
+
+    private void validateBindingMessage(CibaAuthenticationRequest request) {
+        if (!StringUtils.isEmpty(request.getBindingMessage()) && request.getBindingMessage().length() > this.bindingMessageMaxLength) {
+            throw new InvalidBindingMessageException("binding_message is too long");
```

Comment:
Throwing exception instead of calling routingContext.fail(exception) leads to in the log : 

```
09:21:17.127 [vert.x-eventloop-thread-9] [] ERROR io.vertx.ext.web.RoutingContext - Unhandled exception in router
```